test/tests/functional/pbs_dup_acc_log_for_resv.py

```diff
@@ -47,16 +48,439 @@ class TestDupAccLogForResv(TestFunctional):
+                                         reserve_index=rid_instance2)
+        # Restart server second time
+        self.server.restart()
+
+        # Verify accounting log,no duplicate log after server restart
+        m = self.server.accounting_match(
+            msg='.*B;' + rid, id=rid, n='ALL', allmatch=True, regexp=True)
+        self.assertNotEqual(m, None)
+        self.assertEqual(len(m), 2)
+
+    def test_multinode_advance_reservation(self):
+        """
+        Test for duplicate records in accounting log for advance reservations
+        on restart of server requesting multiple nodes
+        """
+        # Submit a advance reservation requesting multinode
+        r = Reservation(TEST_USER)
+        a = {'Resource_List.select': '2:ncpus=1',
+             'Resource_List.place': 'scatter',
+             'reserve_start': int(time.time() + 5),
+             'reserve_end': int(time.time() + 3600)}
+        r.set_attributes(a)
+        rid = self.server.submit(r)
+        a = {'reserve_state': (MATCH_RE, "RESV_RUNNING|5")}
+        self.server.expect(RESV, a, id=rid, offset=6)
+
+        # Verify accounting log before server restart
+        m = self.server.accounting_match(
+            msg='.*B;' + rid, id=rid, n='ALL', allmatch=True, regexp=True)
+        self.assertNotEqual(m, None)
+        self.assertEqual(len(m), 1)
+
+        # Restart server
+        self.server.restart()
+
+        # Verify accounting log, no duplicate log after server restart
+        m = self.server.accounting_match(
+            msg='.*B;' + rid, id=rid, n='ALL', allmatch=True, regexp=True)
+        self.assertNotEqual(m, None)
+        self.assertEqual(len(m), 1)
+
+        a = {'reserve_state': (MATCH_RE, "RESV_RUNNING|5")}
+        self.server.expect(RESV, a, id=rid)
+
+        # Restart server second time
+        self.server.restart()
+        # Verify accounting log, no duplicate log  after server restart
+        m = self.server.accounting_match(
+            msg='.*B;' + rid, id=rid, n='ALL', allmatch=True, regexp=True)
+        self.assertNotEqual(m, None)
+        self.assertEqual(len(m), 1)
+
+    def test_multinode_standing_reservation(self):
+        """
+        Test for duplicate records in accounting log for multi node standing
+        reservations on restart of server requesting multi-node
+        """
+        # PBS_TZID environment variable must be set, there is no way to set
+        # it through the API call, use CLI instead for this test
+
+        _m = self.server.get_op_mode()
+        if _m != PTL_CLI:
+            self.server.set_op_mode(PTL_CLI)
+        if 'PBS_TZID' in self.conf:
+            tzone = self.conf['PBS_TZID']
+        elif 'PBS_TZID' in os.environ:
+            tzone = os.environ['PBS_TZID']
+        else:
+            self.logger.info('Missing timezone, using America/Los_Angeles')
+            tzone = 'America/Los_Angeles'
+        if _m == PTL_API:
+            self.server.set_op_mode(PTL_API)
```

Comment:
Not needed, since this is requires for two specific test related to standing reservation 
For other three test its not needed so I setting in test level only, for the test where PBS_TZID is require , not to setUp()